pkg/ebpf/tracee.go

```diff
@@ -897,57 +897,54 @@ func (t *Tracee) computeConfigValues() []byte {
 	return configVal
 }
 
-// validateKallsymsDependencies load all symbols required by events dependencies
-// from the kallsyms file to check for missing symbols. If some symbols are
-// missing, it will cancel their event with informative error message.
-func (t *Tracee) validateKallsymsDependencies() {
-	var reqKsyms []string
-	symsToDependentEvents := make(map[string][]events.ID)
-	for id := range t.eventsState {
-		for _, symDep := range events.Core.GetDefinitionByID(id).GetDependencies().GetKSymbols() {
-			reqKsyms = append(reqKsyms, symDep.GetSymbol())
-			if symDep.IsRequired() {
-				symEvents, ok := symsToDependentEvents[symDep.GetSymbol()]
-				if ok {
-					symEvents = append(symEvents, id)
-				} else {
-					symEvents = []events.ID{id}
-				}
-				symsToDependentEvents[symDep.GetSymbol()] = symEvents
+func (t *Tracee) checkUnavailableKSymbols(eventsState map[events.ID]events.EventState) map[events.ID][]string {
+	reqKSyms := []string{}
+	kSymbolsToEvents := make(map[string][]events.ID)
+	unavailableKSymsToEvents := make(map[events.ID][]string)
```

Comment:
nit: it's not accurate to call it "...ToEvents" since the map value is an array of symbols (and not events)

Reply:
Changing it to `unavailableKSymsForEventID`.

Reply:
SGTM